bindings/rust/s2n-tls/src/testing/s2n_tls.rs

```diff
@@ -530,11 +529,51 @@ mod tests {
 
         let mut builder = crate::config::Builder::new();
         builder.set_security_policy(&security::DEFAULT_TLS13)?;
-        builder.set_verify_host_callback(AcceptAllHostnames {})?;
+        builder.set_verify_host_callback(UnsecureAcceptAllClientCertificatesHandler {})?;
         builder.load_pem(&fs::read(&cert)?, &fs::read(&key)?)?;
         builder.trust_location(Some(&cert), None)?;
 
-        s2n_tls_pair(builder.build()?);
+        establish_connection(builder.build()?);
+        Ok(())
+    }
+
+    #[test]
+    fn reject_verify_host() -> Result<(), Error> {
```

Comment:
nit: I would name this to something like `connection_level_verify_host` (or something to that effect). the current name isn't clear about its goal.